### DISCUSSION TITLE
"Terraform Authentication on AKS" article issues

### DISCUSSION THREAD

**Discussion by alexazarh** [QualiTorque/qualitorque.github.io#118] [Documentation requests] [answered]
Step 1 Update Azure cli: "az update" results in the following error:
![image](https://user-images.githubusercontent.com/6730546/196213298-f07a3188-0b0d-40ec-9b97-ca70b5175549.png)
Should be "az upgrade"

---
Bullet no. 4 is not correct. 
"If you have an existing AKS then update it with Azure CNI:" the command that follows will not add CNI to the cluster. It will only enable the "pod-identity" feature. Probably need to have a separate bullet for "pod-identity" and for CNI. Also CNI is only a recommendation, it's not a hard requirement if I understand correctly.

---
No. 6 misses a step at the end to click on the create button to actually create the role assignment

---
Type in no. 7:
Create a pod identity in the sandbox namespace, if you don’t have one **_the_** choose/create a namespace for the sandboxes:

---
In no. 4 when creating a cluster, I suggest to parameterize the cluster and RG name. And then in no.7 we can use these env vars for the RG and cluster name instead of 
"az aks pod-identity add --resource-group _**myResourceGroup**_ --cluster-name _**myAKSCluster**_  ..."

---

**Comment by Shnitselon** [comment]
@alexazarh 
Thank you so much for the informative feedback.
RE bullet 4: Can you provide the steps and commands you'd like me to put instead?

**Reply by alexazarh** [reply]
The commands are ok, the text needs to be updated to something like:
"If you don’t have AKS, create one with azure CNI:" -> "If you don’t have AKS cluster, create one with azure CNI and pod managed identity feature:"
"If you have an existing AKS, update it with Azure CNI:" -> "If you have an existing AKS cluster with Azure CNI, update it to enable the pod managed identity feature:"

**Comment by Shnitselon** [comment, marked as answer]
Thanks a lot @alexazarh 
Updated as requested
